### ISSUE TITLE
bug(tools): elixir interpreter missing from shebang dispatch

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#186]
`src/tools.rs::get_from_interpreter` maps shebang interpreter names to
`LANG` variants:

```rust
fn get_from_interpreter(name: &str) -> Option<LANG> {
    match name {
        "sh" | "bash" | "dash" | "ksh" | "zsh" => Some(LANG::Bash),
        "python" => Some(LANG::Python),
        "perl" => Some(LANG::Perl),
        "lua" | "luajit" => Some(LANG::Lua),
        "php" | "php-cgi" => Some(LANG::Php),
        "node" | "nodejs" => Some(LANG::Javascript),
        "tclsh" | "wish" => Some(LANG::Tcl),
        _ => None,
    }
}
```

There is no arm for `"elixir"` / `"iex"`, so an extensionless Elixir
script with `#!/usr/bin/env elixir` falls through to `None` and
`guess_language` returns no language. Standalone Elixir scripts in
`bin/` (the convention for Mix project executables) are therefore not
analysed when invoked without `--language-type elixir`.

## Repro

```sh
cat > /tmp/foo <<'SCRIPT'
#!/usr/bin/env elixir
IO.puts "hello"
SCRIPT
cargo run -p big-code-analysis-cli --quiet -- metrics -p /tmp/foo
```

The file is skipped (or warned about with `-w`) because no language
is detected.

## Fix

Add an arm:

```rust
"elixir" | "iex" => Some(LANG::Elixir),
```

and add regression tests in `src/tools.rs::tests` mirroring
`guess_language_shebang_falls_through_when_no_extension`.

## Related

- Surfaced during /review of the Ruby-language addition (the same gap
  exists for Ruby and is fixed alongside the Ruby PR — this issue
  tracks the Elixir-equivalent fix separately).

## Resolution

Fixed on `fix/issue-186` (commit c5e1dc6, integrated via `fix/batch-2026-05-14`).

- Added `"elixir" | "iex" => Some(LANG::Elixir)` arm to
  `src/tools.rs::get_from_interpreter`.
- Added regression tests
  `guess_language_shebang_detects_elixir_without_extension` and
  `guess_language_shebang_detects_iex_without_extension` mirroring the
  existing ruby/python shebang coverage.
- No public API change; full workspace test suite passes.

**Comment by dekobon** [comment]
Fixed on `fix/issue-186` (commit c5e1dc6, integrated via `fix/batch-2026-05-14`).

`src/tools.rs::get_from_interpreter` now maps `elixir` and `iex` to
`LANG::Elixir`, so extensionless scripts using `#!/usr/bin/env elixir`
or `#!/usr/bin/env iex` are correctly identified.

Two regression tests added under `tools::tests`:

- `guess_language_shebang_detects_elixir_without_extension`
- `guess_language_shebang_detects_iex_without_extension`

Existing version-suffix handling (`strip_version_suffix`) already
covers `elixir1.18` and similar variants; the `env` parser already
handles full-path interpreters via `basename`.
